client/src/main/java/com/aws/greengrass/cli/commands/TopicCommand.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.cli.commands;
+
+import com.aws.greengrass.cli.adapter.NucleusAdapterIpc;
+import picocli.CommandLine;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+@CommandLine.Command(name = "topic", resourceBundle = "com.aws.greengrass.cli.CLI_messages",
+        subcommands = CommandLine.HelpCommand.class, mixinStandardHelpOptions = true,
+        versionProvider = com.aws.greengrass.cli.module.VersionProvider.class)
+public class TopicCommand extends BaseCommand {
+    private final NucleusAdapterIpc nucleusAdapterIpc;
+
+    @Inject
+    public TopicCommand(NucleusAdapterIpc nucleusAdapterIpc) {
+        this.nucleusAdapterIpc = nucleusAdapterIpc;
+    }
+
+    @CommandLine.Command(name = "pub",
+            description = "Publish message to the specific topic ",
+            mixinStandardHelpOptions = true,
+            versionProvider = com.aws.greengrass.cli.module.VersionProvider.class)
+    public void pub(@CommandLine.Option(names = {"-tn", "--topicname"}, paramLabel = "The name of the topic.", required = true) String topicName,
+                    @CommandLine.Option(names = {"-msg", "--message"}, paramLabel = "The message that is published.", required = true) String message,
+                    @CommandLine.Option(names = {"-mt", "--messagetype"}, paramLabel = "The type of the message(local/mqtt).", required = true) String messageType,
```

Comment:
Let's make this an enum flag, not a flag with a value.

`--iotcore` or `--pubsub` are the options that I want.